File: examples/webhook_verification/verify.py
import hashlib
import hmac


def verify_webhook(payload: bytes, signature: str, secret: str) -> bool:
    expected = "sha256=" + hmac.new(secret.encode(), payload, hashlib.sha256).hexdigest()
    return hmac.compare_digest(expected, signature)


if __name__ == "__main__":
    ok = verify_webhook(b'{"event":"payout.settled"}', "sha256=...", "your-secret")
    print("valid:", ok)

File: examples/webhook_verification/verify.go
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/hex"
	"fmt"
)

// VerifyWebhook checks that a PayStream webhook was signed by your endpoint secret.
func VerifyWebhook(payload []byte, signature, secret string) bool {
	mac := hmac.New(sha256.New, []byte(secret))
	mac.Write(payload)
	expected := "sha256=" + hex.EncodeToString(mac.Sum(nil))
	return hmac.Equal([]byte(expected), []byte(signature))
}

func main() {
	ok := VerifyWebhook([]byte(`{"event":"payout.settled"}`), "sha256=...", "your-secret")
	fmt.Println("valid:", ok)
}
